[Refactor] Inline CSS and JS shift to external files

xprilion opened the pull request:
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [x] Updated Forum E2E tests

### What is the current behavior?
Closes #3871 and #3559 and partially #3859

### What is the new behavior?
Moved as much internal CSS and JS from template files to `forum.css` and `forum.js`. Also removed Forum specific code from `server.js` and `server.css`. 

### Other information?
How did you test : e2e tests

andrewaikens87 commented on site/app/templates/forum/CreatePost.twig:
```diff
@@ -1,28 +1,28 @@
             {% elseif activeThreadAnnouncement %}
 
-                <i class="fas fa-star" style="position:relative; display:inline-block; color:gold; -webkit-text-stroke-width: 1px; -webkit-text-stroke-color: black;" title = "Announcement" aria-hidden="true"></i>
+                <i class="fas fa-star active-thread-announcement-2" title = "Announcement" aria-hidden="true"></i>
```
Can we have more descriptive names here for active-thread-announcement-1/2? I think it makes more sense to have a descriptor rather than a number identifier.

xprilion replied:
Done!

andrewaikens87 commented on site/public/js/forum.js:
```diff
@@ -0,0 +1,1057 @@
+$(function(){
```
We probably should have this within a function (this is for forum categories page correct?). Let's have that page reference this function on load instead of having the load functionality here.

xprilion replied:
implementing this made the categories get checkboxes!

xprilion commented:
@andrewaikens87 I was able to solve some of the things pointed, but some I could not replicate or are not related to this issue. 

- The reply button and view history buttons are missing. => They are visible on my laptop! 
- The down arrow for Show more threads => Probably an unrelated issue due to being invisible on my master as well! 
- The edit categories button is adjacent to the thread title textbox => Fixed!
- The "Categories" in this screenshot is displayed inline. => Fixed!
- It seems like the categories aren't being loaded here. => This was an issue in the master as well. Will get fixed when #3899 gets merged. 
- When the page loads the icons in the thread title display as black. => Fixed
- Attachment links previously didn't display with an underline decoration => Fixed

bmcutler commented:
Andrew were you possibly not in hierarchical mode?  reply buttons are not visible in alpha mode, etc.

andrewaikens87 commented:
@bmcutler yes I was, silly mistake. @xprilion please disregard that comment.

andrewaikens87 commented:
Reminder, @xprilion please include the h4 issue fix in this PR that we discussed earlier today. Thanks!

xprilion commented:
@andrewaikens87 I have incorporated all requested changes including the one about the javascript in the Categories Form page! Also, I changed the heading level of the displayed post on the right, from H3 to H2, to make the page as WAVE compatible as possible. It removes the warning of skipped header levels!

andrewaikens87 commented:
Looks like (prior to my merge commit) e2e tests are failing here. Seems like there is an error finding the thread titles. Here is the trace:

Traceback (most recent call last):
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 199, in test_basic_operations_thread
    self.view_thread(title)
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 109, in view_thread
    assert len(thread_title) > 0
AssertionError
======================================================================
FAIL: test_categories (e2e.test_forum.TestForum)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 275, in test_categories
    self.delete_thread(title1)
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 154, in delete_thread
    self.view_thread(title)
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 109, in view_thread
    assert len(thread_title) > 0
AssertionError
======================================================================
FAIL: test_forum_merge_thread (e2e.test_forum.TestForum)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 230, in test_forum_merge_thread
    self.merge_threads(title1, None)
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 168, in merge_threads
    self.view_thread(child_thread_title)
  File "/home/travis/build/Submitty/Submitty/tests/e2e/test_forum.py", line 109, in view_thread
    assert len(thread_title) > 0
AssertionError

xprilion commented:
@andrewaikens87 I guess that might be due to the changing of the element tags. I slept after the commit. Will check it now! :)

andrewaikens87 commented:
Please resolve merge conflicts.

bmcutler commented:
@xprilion @andrewaikens87 
Please one of you should fix the merge conflicts -- I don't want to guess and make a mistake :)

andrewaikens87 commented:
Conflicts resolved, just want to double check locally and will update this comment

andrewaikens87 commented:
Should be good to go!
